toolchain/check/node_stack.h

```diff
@@ -363,91 +461,109 @@ class NodeStack {
   };
   static_assert(sizeof(Entry) == 8, "Unexpected Entry size");
 
+  // Translate a parse node category to the enum ID kind it should always
+  // provide, if it is consistent.
+  static constexpr auto ParseNodeCategoryToIdKind(Parse::NodeCategory category)
```

Comment:
It's not obvious to me what `category` means here in the case where it's not just a single bit: does it mean that we want an `IdKind` that works for anything with an intersection with `category`, or an `IdKind` that works for anything whose category is a superset of `category`?

If the former, then I think we should also return `std::nullopt` for categories including bits outside the five we can handle, because (eg) if we're asked for an `Expr | Whatever`, and it's actually just a `Whatever`, it might not have an `instId`.